src/c/core/session/session.c

```diff
@@ -392,13 +392,18 @@ bool uxr_run_session_until_confirm_delivery(
 {
     UXR_LOCK_SESSION(session);
 
+    int64_t start_timestamp = uxr_millis();
+    int remaining_time = timeout_ms;
+
     uxr_flash_output_streams(session);
 
-    bool timeout = false;
-    while (!uxr_output_streams_confirmed(&session->streams) && !timeout)
+    do
     {
-        timeout = !listen_message_reliably(session, timeout_ms);
+        listen_message_reliably(session, remaining_time);
```

Comment:
Bug discovered:
- If only best effort streams are flashed, `listen_message_reliably(session, remaining_time);` will run for the full function timeout

We should check for `uxr_output_streams_confirmed(&session->streams)` before the do-while beggins

Reply:
Solved with https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/305